crates/uv-pypi-types/src/identifier.rs

```diff
@@ -87,6 +88,39 @@ impl<'de> serde::de::Deserialize<'de> for Identifier {
     }
 }
 
+impl Serialize for Identifier {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        Serialize::serialize(&self.0, serializer)
+    }
+}
+
+#[cfg(feature = "schemars")]
+impl schemars::JsonSchema for Identifier {
+    fn schema_name() -> String {
+        "Identifier".to_string()
+    }
+
+    fn json_schema(_gen: &mut schemars::r#gen::SchemaGenerator) -> schemars::schema::Schema {
+        schemars::schema::SchemaObject {
+            instance_type: Some(schemars::schema::InstanceType::String.into()),
+            string: Some(Box::new(schemars::schema::StringValidation {
+                // Best-effort Unicode support (https://stackoverflow.com/a/68844380/3549270)
+                pattern: Some(r"^[_\p{Alphabetic}][_0-9\p{Alphabetic}]*$".to_string()),
```

Comment:
I didn't find a good resource for proper ECMA script regex validation.

Reply:
cc @Gankra perhaps you could review this part

Reply:
This syntax seems to do the validation you want. You can test this out in e.g. vscode with the "even better toml" extension with the following `uv.toml`s.

This one errors:

```toml
"$schema" = "https://raw.githubusercontent.com/astral-sh/uv/f6e2bbdec7b53a6ce380e9ec27566e88a9144eda/uv.schema.json"

[build-backend]
module-name = "0abc"
```

This one passes:

```toml
"$schema" = "https://raw.githubusercontent.com/astral-sh/uv/f6e2bbdec7b53a6ce380e9ec27566e88a9144eda/uv.schema.json"

[build-backend]
module-name = "abc123"
```

(For local testing you can use `"$schema" = "file://./uv.schema.json"`)